sky/utils/ux_utils.py

```diff
@@ -28,7 +30,11 @@ def print_exception_no_traceback():
             if error():
                 raise ValueError('...')
     """
-    original_tracelimit = getattr(sys, 'tracebacklimit', 1000)
-    sys.tracebacklimit = 0
-    yield
-    sys.tracebacklimit = original_tracelimit
+    if env_options.Options.SHOW_DEBUG_INFO.get():
+        # When SKYPILOT_DEBUG is set, show the full traceback
+        yield
+    else:
```

Comment:
Let's add a comment below L36 on saying that this portion suppresses the traceback or update the description on L22 since this function now has an option of showing the traceback as well.

Haven't tested yet.

Reply:
Thanks! I've updated the docstr. Not updating L37 comment since I feel the `else` condition is self-explanatory.